### PR TITLE
WFLY-14468 Add support for using Infinispan APIs from a deployment

### PR DESCRIPTION
**Pull request by rhusar**
Resolves 
https://issues.redhat.com/browse/WFLY-14468

### CODE REVIEW

**Comment by rhusar** [comment]
@pferraro Please review.

**Review by pferraro — APPROVED** [review]

**Comment by bstansberry** [comment]
I just stopped CI on this to redirect resources to more time-critical jobs. This one gains little from being in WF 23 Beta so it's lower priority until that's tagged.

**Comment by rhusar** [comment]
Sure, that's fine – I have done a push since so will stop those as well.

File: docs/src/main/asciidoc/_high-availability/subsystem-support/Infinispan_Subsystem.adoc [hunk 1]
```diff
@@ -20,14 +20,22 @@ e.g. To create a new cache container that loads marshallers from the "org.bar" m
...
 ----
 
+A cache container can be injected directly into Jakarta EE applications using the `@Resource` annotation, e.g.
+
+[source,java,options="nowrap"]
+----
+@Resource(lookup = "java:jboss/infinispan/container/foo")
+private org.infinispan.manager.EmbeddedCacheManager container;
+----
+
```
Comment (pferraro) [inline docs/src/main/asciidoc/_high-availability/subsystem-support/Infinispan_Subsystem.adoc:38]: This should mention that any caches created from this cache manager will not have a server managed lifecycle, thus the application will have to manage the cache lifecycle manually.

File: docs/src/main/asciidoc/_high-availability/subsystem-support/Infinispan_Subsystem.adoc [hunk 2]
```diff
@@ -445,6 +453,19 @@ e.g. To configure a non-blocking state transfer:
...
+A cache can be injected directly into Jakarta EE applications using the `@Resource` annotation, e.g.
+
+[source,java,options="nowrap"]
+----
+@Resource(lookup = "java:jboss/infinispan/cache/foo/bar") // <1>
+private org.infinispan.Cache<Integer, Object> cache;
+----
+<1> where `foo` is the name of the cache container and `bar` is the name of the cache to inject
+
+
```
Comment (pferraro) [inline docs/src/main/asciidoc/_high-availability/subsystem-support/Infinispan_Subsystem.adoc:468]: Additionally, `@Resource(lookup = "java:jboss/infinispan/cache/foo/default")` will inject the *default* cache of the "foo" container.

**Review by pferraro — CHANGES_REQUESTED** [review]

**Pull request merged**